src/trace_processor/importers/proto/gpu_event_parser.cc

```diff
@@ -123,7 +125,28 @@ GpuEventParser::GpuEventParser(TraceProcessorContext* context)
 
 void GpuEventParser::ParseGpuCounterEvent(int64_t ts, ConstBytes blob) {
   protos::pbzero::GpuCounterEvent::Decoder event(blob.data, blob.size);
-
+  if(!event.has_counter_descriptor()){
+    for (auto it = event.counters(); it; ++it) {
+        // Create a track from descriptorless counter
+        protos::pbzero::GpuCounterEvent_GpuCounter::Decoder counter(*it);
+        if (gpu_counter_track_ids_.find(counter.counter_id()) == gpu_counter_track_ids_.end()) {
+          char numstr[21];
+          char name[] = "gpu_counter ";
+          sprintf(numstr, "%d", counter.counter_id());
+          strcat(name, numstr);
```

Comment:
little memory corruption here

Reply:
can this use std::stringstream instead? not sure if there is a specific perfetto reason for the c style string manipulation here?